sirepo/template/raydata.py

```diff
@@ -63,38 +62,29 @@
     ),
 )
 
-
-def background_percent_complete(report, run_dir, is_running):
-    def _png_filenames():
-        return [
-            pkio.py_path(f).basename for f in sorted(
-                glob.glob(str(run_dir.join('*.png'))),
-                key=os.path.getmtime
+def analysis_job_output_files(data):
+    def _filename_and_image(path):
+        return PKDict(
+            filename=path.basename,
+            image=pkcompat.from_bytes(
+                base64.b64encode(
+                    pkio.read_binary(path),
+                ),
             )
-        ]
+        )
 
-    def _sanitized_name(filename):
-        return sirepo.util.sanitize_string(filename) + 'Animation'
+    def _paths():
+        d = _dir_for_scan_uuid(_parse_scan_uuid(data))
+        for f in sorted(glob.glob(str(d.join('*.png'))), key=os.path.getmtime):
+            yield pkio.py_path(f)
 
-    res = PKDict(
-        pngOutputFiles=[
-            PKDict(name=_sanitized_name(f), filename=f) for f in _png_filenames()
-        ],
-    )
-    res.pkupdate(frameCount=len(res.pngOutputFiles))
-    if is_running:
-        return res.pkupdate(percentComplete=0)
-    return res.pkupdate(percentComplete=100)
+    return PKDict(data=list(map(_filename_and_image, _paths())))
```

Comment:
Why not use a comprehension: `data=[_filename_and_image(x) for x in _paths()]`

Reply:
b9df0c573